lib/ClusterShell/Topology.py

```diff
@@ -456,7 +456,17 @@ def _build_graph(self):
         """
         self.graph = TopologyGraph()
         for src, dst in self._topology:
-            self.graph.add_route(NodeSet(src), NodeSet(dst))
+            # GH#527: router and destination node sets may use NodeSet groups
+            # but they have to resolve to something when we build the graph!
```

Comment:
I'm wondering if this is really a big deal. Couldn't we just ignore those routes? If one of src or dst is empty, just don't `add_route()` ?

Reply:
Great point. It might be more flexible to just ignore such route, I will change that, thanks!